pkg/registry/generic/etcd/etcd.go

```diff
@@ -45,7 +48,31 @@ type Etcd struct {
 	KeyRootFunc func(ctx api.Context) string
 
 	// Called for Create/Update/Get/Delete
-	KeyFunc func(ctx api.Context, id string) (string, error)
+	KeyFunc func(ctx api.Context, name string) (string, error)
+
+	// Called to get the name of an object
+	ObjectNameFunc func(obj runtime.Object) (string, error)
+
+	// Return the TTL objects should be persisted with. Update is true if this
+	// is an operation against an existing object.
+	TTLFunc func(obj runtime.Object, update bool) (uint64, error)
+
+	// Called on all objects returned from the underlying store, after
+	// the exit hooks are invoked.
```

Comment:
It's for pod status, see https://github.com/smarterclayton/kubernetes/commit/aa9df6a4fef507d85cfe6c0dbda8680d3bdc1cb8#diff-069492687bcccc75f3afb6292e6b19e7R75

----- Original Message -----
> > @@ -45,7 +48,31 @@ type Etcd struct {
> >  	KeyRootFunc func(ctx api.Context) string
> >  
> >  	// Called for Create/Update/Get/Delete
> > -	KeyFunc func(ctx api.Context, id string) (string, error)
> > +	KeyFunc func(ctx api.Context, name string) (string, error)
> > +
> > +	// Called to get the name of an object
> > +	ObjectNameFunc func(obj runtime.Object) (string, error)
> > +
> > +	// Return the TTL objects should be persisted with. Update is true if
> > this
> > +	// is an operation against an existing object.
> > +	TTLFunc func(obj runtime.Object, update bool) (uint64, error)
> > +
> > +	// Called on all objects returned from the underlying store, after
> > +	// the exit hooks are invoked.
> 
> An example of intended usage would help. I'd like to avoid usage of this for
> populating fields of the internal representation, for instance. That should
> be done in conversion instead.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/4419/files#r24680829